src/components/academy/index.tsx

```diff
@@ -28,8 +33,18 @@ export const Academy: React.SFC<AcademyProps> = props => (
   </div>
 )
 
-const checkLoggedIn = (props: AcademyProps) =>
-  props.token === undefined ? <Route component={redirectToLogin} /> : undefined
+const checkLoggedIn = (props: IAcademyProps) => {
+  const token = qs.parse(props.location.search).token
+  if (token !== undefined) {
+    props.changeToken(token) // just received a callback from IVLE
```

Comment:
I think this call could use an API call (without further redirect) to check the validity of the token. Redirection is a possible security threat. For example, (`http://localhost/academy?token=SPONGEBOBSQUAREPANTS`) creates a valid token in the state, allowing me to view the game (albeit without any relevant content). It is a matter of debate as to *when* the token is to be validated (as discussed in real life @ningyuansg), but adding an extra check here prevents the user from seeing data that is already present in the website (e.g frontend stuff)

Reply:
- API calls take time, so adding more calls reduces performance
- Invalid token will throw an error with `fetch` anyways, in these cases we catch the error, set `session.token` to undefined, and redirect the user to login (perhaps with a toaster error message)
- If the data is already present in our state, the user must have been logged in before, in a valid manner. Otherwise, the data would not have been fetched. There is no way for 'illegitimate' data to be fetched by an invalid token. Thus, I don't believe this is a security risk

You can document this in an issue though

Reply:
Sure. I'm not very clear about such problems myself, it would be nice to get some more input